Xamarin.PropertyEditing/ViewModels/CreateResourceViewModel.cs

```diff
@@ -0,0 +1,395 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class CreateResourceViewModel
+		: NotifyingObject, INotifyDataErrorInfo
+	{
+		public CreateResourceViewModel (IResourceProvider provider, IEnumerable<object> targets, IPropertyInfo property)
+		{
+			if (provider == null)
+				throw new ArgumentNullException (nameof(provider));
+			if (targets == null)
+				throw new ArgumentNullException (nameof(targets));
+			if (property == null)
+				throw new ArgumentNullException (nameof(property));
+
+			this.provider = provider;
+			this.targets = targets.ToArray();
+			this.property = property;
+
+			CreateResourceCommand = new RelayCommand (OnCreateResource, CanCreateResource);
+
+			LoadingTask = RequestSources ();
+		}
+
+		public event EventHandler<DataErrorsChangedEventArgs> ErrorsChanged;
+		public event EventHandler CreateResource;
+
+		public IPropertyInfo Property => this.property;
+
+		public string ResourceKey
+		{
+			get { return this.resourceKey; }
+			set
+			{
+				if (this.resourceKey == value)
+					return;
+
+				this.resourceKey = value;
+				OnPropertyChanged();
+				((RelayCommand)CreateResourceCommand).ChangeCanExecute();
+
+				if (SelectedResourceSource != null)
+					RequestErrorCheck();
+			}
+		}
+
+		public bool IsKeyed
+		{
+			get { return this.isKeyed; }
+			set
+			{
+				if (this.isKeyed == value)
+					return;
+
+				this.isKeyed = value;
+				OnPropertyChanged ();
+				OnPropertyChanged (nameof(IsAppliedToAll));
+			}
+		}
+
+		public bool IsAppliedToAll
+		{
+			get { return !this.isKeyed; }
+			set
+			{
+				if (this.isKeyed == !value)
```

Comment:
Can you add a comment about why we have two different properties using the same underlying field? At first, I thought it was a bad copy-paste.